src/som/interpreter/Interpreter.java

```diff
@@ -218,7 +208,7 @@ private SAbstractObject doReturnNonLocal(final VirtualFrame frame, final SMethod
       // the "sender" will be the surrounding block and not the object
       // that actually sent the 'value' message.
       SBlock blockT = (SBlock) StackUtils.getCurrentArguments(frame)[0];
-      SAbstractObject receiver = StackUtils.getCurrentArguments(context)[0];
+      SAbstractObject receiver = (SAbstractObject) StackUtils.getCurrentArguments(context)[0];
```

Comment:
Could this cast be moved into the helper? Would it make sense to have the helper be `getCurrentArgument(frame, idx)`?